### DISCUSSION TITLE
Please enable the wiki for taffy repo

### DISCUSSION THREAD

**Discussion by coderedart** [DioxusLabs/taffy#735] [Ideas]
Right now, if you want to use taffy, you need to start by learning css using resources that are all spread out and the quality varies wildly between them. I was hoping that we could have a guide in the wiki that is focused on just taffy. If the wiki is enabled, it can be a common resource which everyone can refer (and contribute) to. It can also contain some common patterns and FAQ for helping integrations.

A couple of useful tips would be:
1. How to interpret the layout struct. 
	1. location is relative to parent
	2. location + size is the whole "area" of this node including margin + border + padding + content
	3. If you shrink whole area with respective lengths from margin rect, you get the area including border + padding + content.
	4. shrink with border rect, to get padding + content
	5. finally, shrink with padding to get content area rect, which is not to be confused with content size which can be overflow beyond bounds.
2. How to enable "transformations" like rotation, scaling and [perspective](https://www.youtube.com/watch?v=pYu3DiWYtL0) among different nodes.
	1. Make a specific "transform node" which is to be treated as a root node. 
	2. If this node is the child of another node, then treat this as a leaf node from the parent's perspective, while the node itself will still use the root measure function.
	3. All widgets in the hierarchy belonging to a certain transform node will share the same coordinate space. 
	4. coordinates need to be transformed going into or out of these transform nodes.

There can be other (better) ways, but having a structured guide in the wiki would be way more helpful. This will not replace github discussions. Instead, content from useful discussion threads can be extracted into the wiki to make it more discoverable.

**Comment by alice-i-cecile** [comment]
I've generally had pretty negative experiences with Github's wiki. These are all really valuable questions though, and I'd like to see them documented within Taffy's code and examples.
